pkg/core/block_base.go

```diff
@@ -80,13 +80,13 @@ func (b *BlockBase) EncodeBinary(bw *io.BinWriter) {
 	b.Script.EncodeBinary(bw)
 }
 
-// createHash creates the hash of the block.
+// CreateHash creates the hash of the block.
 // When calculating the hash value of the block, instead of calculating the entire block,
 // only first seven fields in the block head will be calculated, which are
 // version, PrevBlock, MerkleRoot, timestamp, and height, the nonce, NextMiner.
 // Since MerkleRoot already contains the hash value of all transactions,
 // the modification of transaction will influence the hash value of the block.
-func (b *BlockBase) createHash() error {
+func (b *BlockBase) CreateHash() error {
```

Comment:
This one seems to be wrong. You're not supposed to trigger this one from the outside. I see that you're exporting it for the `newBlock()` in testing code, but it probably can be solved in some other fashion.